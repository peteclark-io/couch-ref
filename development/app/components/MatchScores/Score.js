'use strict'

import React from 'react';
import styles from './styles.css';

import GoalCount from './GoalCount';
import Team from './Team';
import KickOff from './KickOff';

const Score = React.createClass({

   propTypes: {
      match: React.PropTypes.shape({
         'id': React.PropTypes.string,
         'kickOff': React.PropTypes.object,
         'home': React.PropTypes.string,
         'away': React.PropTypes.string,
         'goalsHome': React.PropTypes.number,
         'goalsAway': React.PropTypes.number
      })
   },

   render: function() {
      return (
         <div ref={node => (this.root = node)}>
            <KickOff kickOff={this.props.match.kickOff} />
            <div className={styles.score}>
               <Team name={this.props.match.home} />
               <GoalCount goals={this.props.match.goalsHome} />
            </div>
            <div className={styles.score}>
               <Team name={this.props.match.away} />
               <GoalCount goals={this.props.match.goalsAway} />
            </div>
         </div>
      );
   }
});

export default Score;
