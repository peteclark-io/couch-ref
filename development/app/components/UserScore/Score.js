'use strict'

import React from 'react';
import {connect} from 'react-redux'
import classNames from 'classnames';

import styles from './styles.css';

export const Score = React.createClass({

   propTypes: {
      rank: React.PropTypes.number,
      movement: React.PropTypes.number
   },

   render: function() {
      return (
         <div className={classNames(styles.container, styles.centred)}>
            <h1 className={styles.heading}>Your Ranking!</h1>
            <h2 className={styles.score}>#{this.props.rank ? this.props.rank : 'Unranked!'}</h2>
         </div>
      );
   }
});

const getRank = (state = {user: {}}) => {
   return state.user.rank;
};

const getMovement = (state = {user: {}}) => {
   return state.user.movement;
};

const mapStateToProps = (state) => {
   return {
      rank: getRank(state),
      movement: getMovement(state)
   }
};

const LiveScore = connect(
   mapStateToProps
)(Score);

export default LiveScore;
