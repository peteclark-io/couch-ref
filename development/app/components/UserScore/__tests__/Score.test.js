'use strict';

import React from 'react';
import {Score} from '../Score';
import renderer from 'react-test-renderer';
import {mount, shallow} from 'enzyme';

it('Renders score ok.', () => {
   const rendered = renderer.create(
      <Score score={4000.1123} />
   );

   expect(rendered.toJSON()).toMatchSnapshot();
});
