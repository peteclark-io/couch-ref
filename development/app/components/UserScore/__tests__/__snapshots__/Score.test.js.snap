exports[`test Renders score ok. 1`] = `
<div
  className="container centred">
  <h1
    className="heading">
    Your Score!
  </h1>
  <h2
    className="score">
    4000
  </h2>
</div>
`;
